internal/services/bot/bot_channel_web_chat_resource.go

```diff
@@ -53,16 +54,72 @@ func resourceBotChannelWebChat() *pluginsdk.Resource {
 				ValidateFunc: validate.BotName,
 			},
 
-			"site_names": {
+			"site": {
 				Type:     pluginsdk.TypeSet,
-				Required: true,
-				Elem: &pluginsdk.Schema{
-					Type:         pluginsdk.TypeString,
-					ValidateFunc: validation.StringIsNotEmpty,
+				Optional: true,
+				Computed: !features.FourPointOhBeta(),
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"application_id": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"block_user_upload_enabled": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+
+						"endpoint_parameters_enabled": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+
+						"no_storage_enabled": {
```

Comment:
Same here, this should be renamed to 
```suggestion
						"storage_enabled": {
```

Reply:
Updated

Reply:
I assume we may not rename it to "storage_enabled" because storage_enabled has different meaning than no_storage_enabled.

When no_storage_enabled is true, it means the site without storage is added.
When storage_enabled is true, it means the site with storage is added.

Reply:
I renamed it to "storage_enabled".